Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -393,18 +393,12 @@ inline void WidthIterator::advanceInternal(TextIterator& textIterator, GlyphBuff
 
     auto glyphData = m_font->glyphDataForCharacter(character, false, FontVariant::NormalVariant);
 
-    auto getHalfWidthFontAndSetGlyphDataIfNeeded = [&](GlyphData& glyphData, const TextSpacing::CharactersData& charactersData) {
-        auto halfWidthFont = fontDescription.textSpacingTrim().shouldTrimSpacing(charactersData) ? glyphData.font->halfWidthFont() : nullptr;
-        if (halfWidthFont)
-            glyphData.font = halfWidthFont.get();
-        return halfWidthFont;
-    };
-
     RefPtr<Font> halfWidthFont;
     auto shouldProcessTextSpacingTrim = !fontDescription.textSpacingTrim().isSpaceAll();
     if (shouldProcessTextSpacingTrim) {
         TextSpacing::CharactersData charactersData = { .currentCharacter = character, .currentCharacterClass = TextSpacing::characterClass(character) };
-        halfWidthFont = getHalfWidthFontAndSetGlyphDataIfNeeded(glyphData, charactersData);
+        halfWidthFont = TextSpacing::getHalfWidthFontIfNeeded(*glyphData.font, fontDescription.textSpacingTrim(), charactersData);
+        glyphData.font = halfWidthFont ? halfWidthFont.get() : glyphData.font;
```

Comment:
Maybe this should just be an `if` since the second result in the ternary is just the object itself?